src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -212,52 +202,40 @@ internal static int Encrypt(SafeSslHandle context, byte[] input, int offset, int
                     case Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN:
                     case Ssl.SslErrorCode.SSL_ERROR_WANT_READ:
                         break;
-
+                    //indicates we need to write the out buffer and write again
+                    case Ssl.SslErrorCode.SSL_ERROR_WANT_WRITE:
+                        break;
                     default:
                         throw new SslException(SR.Format(SR.net_ssl_encrypt_failed, errorCode), innerError);
                 }
             }
-            else
-            {
-                int capacityNeeded = Crypto.BioCtrlPending(context.OutputBio);
-
-                if (output == null || output.Length < capacityNeeded)
-                {
-                    output = new byte[capacityNeeded];
-                }
-
-                retVal = BioRead(context.OutputBio, output, capacityNeeded);
-            }
-
-            return retVal;
+            return context.OutputBio.TakeBytes(out output);
```

Comment:
I'm confused because this is an `out` which replaces `output`.  `output` was only `ref` in this method in case the caller under-allocated for the frame.

If the common code, which calls into the PAL, is renting a pooled buffer, and we're outing a different pooled buffer, are we returning all the leased objects?

Maybe I'm misunderstanding where the input version of `output` comes from?